lib/suppressions.cpp

```diff
@@ -169,8 +169,20 @@ std::vector<Suppressions::Suppression> Suppressions::parseMultiSuppressComment(c
 
 std::string Suppressions::addSuppressionLine(const std::string &line)
 {
-    std::istringstream lineStream(line);
+    std::istringstream lineStream;
     Suppressions::Suppression suppression;
+
+    // Strip any end of line comments
+    std::string::size_type endpos = std::min(line.find("#"), line.find("//"));
```

Comment:
this means that if the filename contains `#` then you cant suppress stuff in that file. however it seems a bit unlikely

Reply:
Double slashes is probably more likely.  `:` is already going to be a problem.  Checking for whitespace would probably be worse, as I guess that would be more likely, especially on windows.

Reply:
> Double slashes is probably more likely

Are you talking about uris? I have the feeling that a uri such as `file:///x.cpp` won't work. unless `std::ifstream f("file:///x.cpp");` would work.

The `:` in the filename is not a problem as far as I know. But the error id can't have that.

Reply:
I was thinking more about windows shares like \\server\path\file being written with forward slashes. Not sure how relevant that is. 
I haven't worked in a windows environment for many years...

Reply:
yes I guess that cannot be ruled out. I have not tried to run on some share. well it seems unlikely both with `//` and with `#` and I guess people can suppress with for instance xml if they need to suppress these anyway.